framework/src/utils/BootstrapStatistics.C

```diff
@@ -46,7 +46,15 @@ makeBootstrapCalculator(const MooseEnum & item,
   ptr->setSeed(seed);
   if (!ptr)
     ::mooseError("Failed to create Statistics::BootstrapCalculator object for ", item);
+
+    // The minimum compiler GCC 4.8.4 needs the std::move explicitly; the newer compilers do this
+    // automatically.
+#if defined(__GNUC__) && (__GNUC__ == 4) && (__GNUC_MINOR__ < 9) && !defined(__INTEL_COMPILER) &&  \
+    !defined(__clang__)
+  return std::move(ptr);
```

Comment:
I just checked with the GCC 4.8.4 compiler. You definitely do not need this line (I tested it on rod). The issue is that the type of your return and the type you declared are mismatched. We have a few places in the framework that return unique_ptrs just fine without move. Here's one:   https://github.com/idaholab/moose/blob/5fa3664e48b3d1755ce1b2f050ed276b8f269327/framework/src/meshgenerators/FancyExtruderGenerator.C#L565

Reply:
Interesting, I wonder why the std::move fixes this issue then? I will fix the type mis-match and see what happens.